add Queue-Times API Example

DJDevon3 opened the pull request:
Used for Disney theme park queue times. Public API.  Runs infinitely until time to update.  The way this one works is slightly different from most examples.  It's designed for a small display to take to the theme park and get updates on queue times.  

Here's an [example on a display](https://github.com/DJDevon3/My_Circuit_Python_Projects/tree/main/Boards/espressif/Unexpected%20Maker%20Feather%20S3/4.0%20ST7796S/QueueTimes%20API%20Display).

Inspired by Reddit user [Fishing_Quiet asked for help getting this API working on a display](https://www.reddit.com/r/circuitpython/comments/1bkpynl/updating_json_file/).

FoamyGuy commented on examples/wifi/expanded/requests_wifi_api_queuetimes.py:
```diff
@@ -0,0 +1,106 @@
+                qtimes_queuetime = str(ride["wait_time"])
+                qtimes_isopen = str(ride["is_open"])
+
+                print(f" |  | Rides: {qtimes_rides}")
```
It's very minor, but I think "Ride" or "Ride Name" would be better than "Rides" plural with the s on the end. 

I'm not that familiar with Disney rides but they looked like singular ride names in the list in the ones that I saw.

DJDevon3 replied:
Will do

FoamyGuy reviewed (changes requested):
Neat, thanks for publishing this!

I think I'm in favor of just printing the results once instead of looping to print the same data over and over. 

That and one other tweak are noted with inline comments.

DJDevon3 commented on examples/wifi/expanded/requests_wifi_api_queuetimes.py:
```diff
@@ -0,0 +1,106 @@
+        time.sleep(60)
+        break
+
+    # Loop infinitely until its time to re-poll
```
@FoamyGuy Yes, that is specifically the way they requested it to behave.  

I suppose printing a "last updated" timestamp would make sense but would take up precious screen real estate.  It makes less sense on a huge 3.5" TFT.  

The project intention is for a portable SSD1306 with a lipo battery on a watch band.  In the real script it will attempt to reconnect (via a tethered phone hotspot) every 1 minute.  Perhaps I could improve upon the code comments to specify that's the intention of this particular example?

If you gave one device to each kid and it only worked within range of the hotspot it could ensure that your kids will want to stay close enough to ensure the device works so they get ride updates too.  It's a brilliant incentive for your kids to want to stay close to you in a theme park.

Maybe change the text color after an update so that every new update is unmistakably new?  That would serve as an indicator of new content without taking up any screen real estate.

FoamyGuy replied:
I am still in favor of removing the looping behavior from the example here. That sounds like a neat project and they can definitely still use it that way if they'd like. I could also see a world where someone might want to "favorite" certain rides to get information about them more prominently or even some alerts based on the data returned. 

I believe that in it's usage as an example here the continued printing of the same data over and over makes it more confusing because it's difficult to tell what is happening and when there is actually new data. None of the rest of the examples in this repo behave that way, they're all either "single shot" fetching once and ending, or repeating in a loop that fetches every X minutes or seconds and prints the current data then waits until it's time to run again. I believe it's best to have this example match that behavior and only print the data once after it's been fetched.

DJDevon3 replied:
@FoamyGuy sounds good.  I have the example that the person wanted in my personal repo, and linked them there, and they already have that one.  Turning it into a single use "code done running" sounds good to me.

FoamyGuy commented:
I think removing the extra infinite looping of the printouts when there isn't new data is the only thing left to change on this one. Everything else is looking good to me at this point.

If you'd prefer I can work on that change and commit it here, let me know.

DJDevon3 commented:
I missed working on this one last night sorry!  If you would like to do it go for it.  I have bursted irrigation pipes to deal with for the next couple of days.  Not a water main but something that will cost me money if I don't address immediately.

justmobilize commented on examples/wifi/expanded/requests_wifi_api_queuetimes.py:
```diff
@@ -0,0 +1,106 @@
+
+    try:
+        print(" | Attempting to GET Queue-Times JSON!")
+        try:
+            qtimes_response = requests.get(url=QTIMES_SOURCE)
+            qtimes_json = qtimes_response.json()
+        except ConnectionError as e:
+            print("Connection Error:", e)
+            print("Retrying in 10 seconds")
+        print(" | ✅ Queue-Times JSON!")
+
+        DEBUG_QTIMES = False
+        if DEBUG_QTIMES:
+            print("Full API GET URL: ", QTIMES_SOURCE)
+            print(qtimes_json)
+        qtimes_response.close()
+        print("✂️ Disconnected from Queue-Times API")
```
```suggestion
        try:
            with requests.get(url=QTIMES_SOURCE) as qtimes_response:
                qtimes_json = qtimes_response.json()
        except ConnectionError as e:
            print("Connection Error:", e)
            print("Retrying in 10 seconds")
        print(" | ✅ Queue-Times JSON!")

        DEBUG_QTIMES = False
        if DEBUG_QTIMES:
            print("Full API GET URL: ", QTIMES_SOURCE)
            print(qtimes_json)
```

DJDevon3 commented:
@FoamyGuy ready for another review.  removed infinite loop and added Justins suggested change to use with.  Cleaned up serial prints to be more legible.

FoamyGuy reviewed (approved):
Thank you!

I committed one final tweak to remove the call to `close()` since the with context handles it now and to move the printout inside of the try block since we won't have any data to print if the connection failed and removed the sleep() calls to make it print them all at once.
